### PR TITLE
terraform: remove cloud loggers

### PR DESCRIPTION
**Pull request by msanft**
### Context

With the [deprecation of Azure's Application Insights](https://teams.microsoft.com/l/message/19:6c4f7544f97340dbb6c4dcd73aa6c205@thread.skype/1707122321437?tenantId=adb650a8-5da3-4b15-b4b0-3daf65ff7626&groupId=62348c88-6944-413d-9232-9d3adb03009e&parentMessageId=1707122321437&teamName=Edgeless&channelName=Constellation&createdTime=1707122321437), we need to migrate away from using them for cloud logs. After a discussion, we concluded to remove cloud logging altogether, as we aggregate such logs in our CI through OpenSearch, and customers can use CSP APIs or deploy their own logging infrastructure to get to the boot logs.

### Proposed change(s)

- Remove cloud logging in the bootstrapper.
- Remove creation of cloud logging resources through Terraform. For GCP, this is a project-wide API enablement, which we don't manage through Terraform, so the PR does not include a change for GCP here.
- Remove the logging endpoint in the QEMU metadata API.
- Overhaul the docs regarding getting to boot logs of Constellation nodes.

> [!IMPORTANT]
> When Azure disallows the creation of new application insights resources, users won't be able to create new clusters with an old Constellation version still relying on said resource. Thus, this change should be included in a release before the resource is abandoned. Existing clusters should not break, given there is a brownout period for existing application insights resources.

### Additional info

- [AB#3847](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/3847), [AB#3781](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/3781)
- [E2E Test](https://github.com/edgelesssys/constellation/actions/runs/7783922106)

### Checklist

- [x] Run the E2E tests that are relevant to this PR's changes
- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

**Review by malt3 — APPROVED** [review]
LGTM

File: docs/docs/workflows/troubleshooting.md [hunk 1]
```diff
@@ -95,49 +95,14 @@ check if the encountered [issue is known](https://github.com/edgelesssys/constel
 
 ## Diagnosing issues
 
-### Cloud logging
+### Boot Logs
```
Comment (thomasten) [inline docs/docs/workflows/troubleshooting.md:98]:
```suggestion
### Logs
```

File: docs/docs/workflows/troubleshooting.md [hunk 2]
```diff
@@ -95,49 +95,14 @@ check if the encountered [issue is known](https://github.com/edgelesssys/constel
...
 
-To provide information during early stages of a node's boot process, Constellation logs messages to the log systems of the cloud providers. Since these offerings **aren't** confidential, only generic information without any sensitive values is stored. This provides administrators with a high-level understanding of the current state of a node.
+To get started on diagnosing issues with Constellation, it often is helpful to collect logs from nodes, pods or other resources in the cluster. Most logs are available through Kubernetes' standard
```
Comment (thomasten) [inline docs/docs/workflows/troubleshooting.md:100]:
```suggestion
To get started on diagnosing issues with Constellation, it's often helpful to collect logs from nodes, pods, or other resources in the cluster. Most logs are available through Kubernetes' standard
```

File: docs/docs/workflows/troubleshooting.md [hunk 3]
```diff
@@ -95,49 +95,14 @@ check if the encountered [issue is known](https://github.com/edgelesssys/constel
...
+[logging interfaces](https://kubernetes.io/docs/concepts/cluster-administration/logging/).
 
-You can view this information in the following places:
+To debug issues occurring at boot time of the nodes, one can use the serial console interface of the CSP.
```
Comment (thomasten) [inline docs/docs/workflows/troubleshooting.md:103]:
```suggestion
To debug issues occurring at boot time of the nodes, you can use the serial console interface of the CSP.
```

**Review by thomasten — APPROVED** [review]

File: docs/docs/workflows/troubleshooting.md [hunk 4]
```diff
@@ -95,49 +95,14 @@ check if the encountered [issue is known](https://github.com/edgelesssys/constel
...
 
-### Cloud logging
+### Logs
 
-To provide information during early stages of a node's boot process, Constellation logs messages to the log systems of the cloud providers. Since these offerings **aren't** confidential, only generic information without any sensitive values is stored. This provides administrators with a high-level understanding of the current state of a node.
+To get started on diagnosing issues with Constellation, it's often helpful to collect logs from nodes, pods, or other resources in the cluster. Most logs are available through Kubernetes' standard
+[logging interfaces](https://kubernetes.io/docs/concepts/cluster-administration/logging/).
 
-You can view this information in the following places:
+To debug issues occurring at boot time of the nodes, you can use the serial console interface of the CSP.
```
Comment (daniel-weisse) [inline docs/docs/workflows/troubleshooting.md:103]: I think this option is actually disabled everywhere where its possible for non debug deployments.

Comment (thomasten) [reply docs/docs/workflows/troubleshooting.md:103]: Also seeing the output? Or only logging in?

Comment (msanft) [reply docs/docs/workflows/troubleshooting.md:103]: Verifying this currently. I think we should have a read-only console on all CSPs

Comment (thomasten) [reply docs/docs/workflows/troubleshooting.md:103]: Ah, you mean the serial console is disabled via the CSP, but not in a CC-secure way, right?

Comment (msanft) [reply docs/docs/workflows/troubleshooting.md:103]: Well, the read-only configuration is embedded in our image and thus also "CC-secure", but the cloud provider may preserve the logs after boot, which we seemingly explicitly turn off where possible. However, the serial console in all cases is read-only and, using our image, doesn't disclose any sensible information, so I don't think it's a problem.

Comment (thomasten) [reply docs/docs/workflows/troubleshooting.md]: Yes, I didn't mean that there's a security issue. I only remembered that, in addition, we may have turned off serial console access via VM settings as a kind of best-effort approach.

**Pull request merged**